common/network-common/src/main/java/org/apache/spark/network/server/TransportRequestHandler.java

```diff
@@ -145,7 +172,12 @@ private void processStreamRequest(final StreamRequest req) {
     }
 
     if (buf != null) {
-      respond(new StreamResponse(req.streamId, buf.size(), buf));
+      respond(new StreamResponse(req.streamId, buf.size(), buf)).addListener(future -> {
+        if (streamManager instanceof OneForOneStreamManager) {
```

Comment:
When executor fetch file from driver, it will send `StreamRequest`. `NettyStreamManager` on driver will serve the stream.